src/main/java/me/dinowernli/grpc/polyglot/command/ServiceList.java

```diff
@@ -20,97 +22,146 @@
 
   /** Lists the GRPC services - filtered by service name (contains) or method name (contains) */
   public static void listServices(
-      Output output,
-      FileDescriptorSet fileDescriptorSet,
-      String protoDiscoveryRoot,
-      Optional<String> serviceFilter,
-      Optional<String> methodFilter,
-      Optional<Boolean> withMessage) {
+    Output output,
+    FileDescriptorSet fileDescriptorSet,
+    String protoDiscoveryRoot,
+    Optional<String> serviceFilter,
+    Optional<String> methodFilter,
+    Optional<Boolean> withMessage,
+    Optional<Boolean> jsonOutput) {
+
+    ArrayList<Service> services = new ArrayList<>(
+      getServices(fileDescriptorSet, protoDiscoveryRoot, serviceFilter, methodFilter, withMessage, jsonOutput));
+
+    if (jsonOutput.isPresent() && jsonOutput.get()) {
+      Gson gson = new Gson();
+      output.writeLine(gson.toJson(services));
```

Comment:
I looked into using the JSON printing but when I tried this it looked like it would take more  work. The JSON printed did not resolve all the way down to the primitive level as the as the output currently does. Say a proto message is defined:
```"foo": {
    "name": "STRING",
    "id": "INT"
},
"bar": "ENUM" ```
The JSON output would be:
```"foo": "Foo",
"bar": "ENUM"```
This doesn't give enough information to people to allow them to construct their own messages.